core/src/main/java/io/ddf/etl/TransformationHandler.java

```diff
@@ -5,77 +5,119 @@
 import com.google.common.collect.Lists;
 import io.ddf.DDF;
 import io.ddf.analytics.Summary;
+import io.ddf.content.Schema;
 import io.ddf.content.Schema.Column;
 import io.ddf.content.Schema.ColumnClass;
 import io.ddf.content.Schema.ColumnType;
 import io.ddf.datasource.SQLDataSourceDescriptor;
 import io.ddf.exception.DDFException;
 import io.ddf.misc.ADDFFunctionalGroupHandler;
 import io.ddf.util.Utils;
+import org.apache.commons.lang.StringEscapeUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.*;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 public class TransformationHandler extends ADDFFunctionalGroupHandler implements IHandleTransformations {
 
+  private static Logger LOG = LoggerFactory.getLogger(TransformationHandler.class);
+
   public TransformationHandler(DDF theDDF) {
     super(theDDF);
     // TODO Auto-generated constructor stub
   }
 
-  @Override public DDF transformScaleMinMax() throws DDFException {
-    Summary[] summaryArr = this.getDDF().getSummary();
-    List<Column> columns = this.getDDF().getSchema().getColumns();
+  @Override
+  public DDF transformScaleMinMax() throws DDFException {
+    return this.transformScaleMinMax(null, Boolean.FALSE);
+  }
 
+  @Override
+  public DDF transformScaleMinMax(List<String> columns, Boolean inPlace) throws DDFException {
+    Summary[] summaryArr = this.getDDF().getSummary();
+    Schema schema = this.getDDF().getSchema();
     // Compose a transformation query
-    StringBuffer sqlCmdBuffer = new StringBuffer("SELECT ");
+    StringBuilder sqlCmdBuffer = new StringBuilder("SELECT ");
 
-    for (int i = 0; i < columns.size(); i++) {
-      Column col = columns.get(i);
-      if (!col.isNumeric() || col.getColumnClass() == ColumnClass.FACTOR) {
-        sqlCmdBuffer.append(col.getName()).append(" ");
+    for (int colIndex = 0; colIndex < schema.getNumColumns(); colIndex++) {
+      Column col = schema.getColumn(colIndex);
+      String escapedColName = StringEscapeUtils.escapeJava(col.getName());
+
+      if (!isColumnEligibleToScale(col, columns)) {
```

Comment:
@ducleminh Should we check for non-existent requested columns so that we don't have to run and return the new DDF with the same content in case of inPlace = false?